akka-docs/src/main/paradox/additional/rolling-updates.md

```diff
@@ -0,0 +1,90 @@
+
+## Cluster Sharding
+
+During a rolling upgrade, sharded entities receiving traffic may be moved during @ref:[shard rebalancing](../cluster-sharding.md#shard-rebalancing), 
+to an old or new node in the cluster, based on the pluggable allocation strategy and settings.
+When an old node is stopped the shards that were running on it may be allocated to one of the
+other old nodes remaining in the cluster. See @ref[ClusterSingleton](#cluster-singleton) for a useful `ShardCoordinator` optimization.
+
+There are some cases when @ref:[a full cluster restart is needed](../cluster-sharding.md#rolling-upgrades).
+
+## Cluster Singleton
+
+It's more efficient to avoid moving a `ClusterSingleton` more than necessary because they typically have to recover their state
+and it might introduce unnecessary delays.
+
+An optional optimization is to leave the oldest node running a `ClusterSingleton` until last
+to avoid it having to move more than once. 
+
+## Migrating Untyped to Typed
+
+It is recommended with a two step approach:
+
+* Deploy with the new nodes set to `akka.cluster.configuration-compatibility-check.enforce-on-join = off`
+and ensure all nodes are in this state
+* Deploy again and with the new nodes set to `akka.cluster.configuration-compatibility-check.enforce-on-join = on`. 
+
+The configuration from existing nodes should pass the @ref:[Cluster Configuration Compatibility Checks](#cluster-configuration-compatibility-check).
+Find out more about coexisting and @ref:[untyped to typed](../typed/coexisting.md#untyped-to-typed). 
+
+### With Cluster Sharding and Persistence
+
+Rolling upgrades where shards on old nodes are running untyped persistent actors 
+and new ones are running typed persistent behaviors have been tested successfully by the team and users.
+Samples coming soon.
+
+## Cluster Shutdown
+ 
+@ref:[Coordinated Shutdown](../actors.md#coordinated-shutdown) will automatically run on SIGTERM when the cluster node sees itself as Exiting.
+Thus running shutdown tasks in a JVM shutdown hook is not recommended.
+@ref:[Graceful shutdown](../cluster-sharding.md#graceful-shutdown) of Cluster Singletons and Cluster Sharding similarly happen automatically.
+ 
+In case of network failures it may still be necessary to set the node’s status to Down in order to complete the removal. 
+
+Find out more about
+* @ref:[Cluster Downing](../cluster-usage.md#downing) and providers
+* [Cluster Bootstrap](https://doc.akka.io/docs/akka-management/current/bootstrap/index.html#rolling-updates) and Rolling updates
+* [Split Brain Resolver](https://doc.akka.io/docs/akka-enhancements/current/split-brain-resolver.html)
+
+## Cluster Configuration Compatibility Checks
+
+Relevant information on rolling updates and enforcing @ref:[Akka Cluster configuration compatibility checks](../cluster-usage.md#configuration-compatibility-check)
```

Comment:
This is linked above as well, can the information be merged / expanded?

Reply:
Yes, done :)